include/soci/oracle/soci-oracle.h

```diff
@@ -288,9 +290,20 @@ struct oracle_blob_backend : details::blob_backend
 
     void trim(std::size_t newLen) SOCI_OVERRIDE;
 
+    locator_t get_lob_locator() const;
+
+    void set_lob_locator(locator_t locator, bool initialized = true);
+
+    void reset();
+
+    void ensure_initialized();
+
+private:
     oracle_session_backend &session_;
 
-    OCILobLocator *lobp_;
+    locator_t lobp_;
+
+    bool initialized_;
```

Comment:
Is `initialized_` different from `lobp_ != NULL`? If so, when does this happen and what does it mean?

Reply:
yes